cli/assets.go

```diff
@@ -256,3 +270,318 @@ func deleteAssetByIDCommand() *cobra.Command {
 
 	return cmd
 }
+
+func listAllTypesCommand() *cobra.Command {
+	var types, services, data, q, q_fields string
+	cmd := &cobra.Command{
+		Use:   "types",
+		Short: "lists all asset types",
+		Example: heredoc.Doc(`
+			$ compass asset types
+			$ compass asset types -t type1 --query query1 --query_fields q_fields1
+		`),
+		Annotations: map[string]string{
+			"action:core": "true",
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			spinner := printer.Spin("")
+			defer spinner.Stop()
+
+			clnt, cancel, err := client.Create(cmd.Context())
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			ctx := client.SetMetadata(cmd.Context())
+			res, err := clnt.GetAllTypes(ctx, makeGetAllTypesRequest(types, services, data, q, q_fields))
+			if err != nil {
+				return err
+			}
+			spinner.Stop()
+
+			report := [][]string{}
+			report = append(report, []string{"NAME", "COUNT"})
+			index := 1
+			for _, i := range res.GetData() {
+				report = append(report, []string{term.Bluef(i.Name), fmt.Sprintf("%v", i.Count)})
+				index++
+			}
+			printer.Table(os.Stdout, report)
+
+			return nil
+		},
+	}
+	cmd.Flags().StringVarP(&types, "types", "t", "", "filter by types")
+	cmd.Flags().StringVarP(&services, "services", "s", "", "filter by services")
+	cmd.Flags().StringVarP(&data, "data", "d", "", "filter by field in asset.data")
+	cmd.Flags().StringVar(&q, "query", "", "filter by specific query")
+	cmd.Flags().StringVar(&q_fields, "query_fields", "", "filter by query field")
+
+	return cmd
+}
+
+func makeGetAllTypesRequest(types, services, data, q, q_fields string) *compassv1beta1.GetAllTypesRequest {
```

Comment:
Nothing specific to this code but I wonder why we have filters on a simple list API of asset types. @sudo-suhas can you help me understand the usecase?